### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by gamache**
This PR started out as "Multiple Queues" but has grown much larger.

I refactored Jorb with the following goals:
* support for reading multiple queues in priority order, instead of one-queue-per-job
* more docs
* better config
* more consistent naming

This release is not at all backward-compatible.

### CODE REVIEW

File: lib/jorb/backend/sqs.ex [hunk 1]
```diff
@@ -2,39 +2,81 @@ defmodule Jorb.Backend.SQS do
...
+      {:ok, _} -> :ok
+      err -> err
+    end
   end
 
-  def purge(queue_name) do
-    with request <- SQS.purge_queue(queue_name),
+  @impl true
+  def enqueue_message(queue, payload, _opts) do
+    with {:ok, encoded} <- Poison.encode(payload),
```
Comment (andyleclair) [inline lib/jorb/backend/sqs.ex:37]: Should we switch to `Jason`?

File: lib/jorb.ex [hunk 2]
```diff
@@ -6,35 +6,117 @@ defmodule Jorb do
...
+    single job, default 5000.
 
   """
 
-  def backend() do
-    Application.get_env(:jorb, :backend)
+  @defaults [
+    backend: Jorb.Backend.Memory,
+    write_interval: 1000,
+    write_batch_size: 1,
```
Comment (andyleclair) [inline lib/jorb.ex:100]: It appears that we're not doing anything with this param. Should this get added or is that for a separate enhancement?

File: lib/jorb/job.ex [hunk 3]
```diff
@@ -1,36 +1,191 @@
...
+        start: {Jorb.Worker, :start_link, [[{:module, module} | opts]]},
+        type: :worker,
+        restart: :permanent,
+        shutdown: 5000
+      }
+    end)
+  end
+
+  @doc ~S"""
+  Returns a list of one or more child specs for GenServers that
```
Comment (andyleclair) [inline lib/jorb/job.ex]: Does this actually? It looks like it just starts reading from the things

Comment (andyleclair) [reply lib/jorb/job.ex]: Unless I don't know what `Task.yield_many` does

**Comment by andyleclair** [comment]
A few comments, but otherwise, this is a great refactor. I'm super pumped to see effort going into this bad boy

File: lib/jorb.ex (hunk 2 above)
Comment (gamache) [reply lib/jorb.ex:100]: I'm doing the Jorb queuer next, this would hook into that.

File: lib/jorb/job.ex (hunk 3 above)
Comment (gamache) [reply lib/jorb/job.ex]: Oh poop, I switched the docs for `work` and `workers`, good catch.

File: lib/jorb/backend/sqs.ex (hunk 1 above)
Comment (gamache) [reply lib/jorb/backend/sqs.ex:37]: Yup, Jason and Mojito, I'll do it next PR.

**Pull request merged**